bouncer/tests/broker_level_screening.ts

```diff
@@ -56,35 +56,82 @@ async function newAssetAddress(asset: InternalAsset, seed = null): Promise<strin
 /**
  * Submits a transaction as tainted to the extrinsic on the state chain.
  *
- * @param txId - The txId to submit as tainted as byte array in the order it is on the Bitcoin chain - which
- * is reverse of how it's normally displayed in block explorers.
+ * @param txId - The txId to submit as tainted, in its typical representation in bitcoin explorers,
+ * i.e., reverse of its memory representation.
  */
-async function submitTxAsTainted(txId: number[]) {
+async function submitTxAsTainted(txId: string) {
+  // The engine uses the memory representation everywhere, so we convert the txId here.
+  const memoryRepresentationTxId = hexStringToBytesArray(txId).reverse();
   await using chainflip = await getChainflipApi();
   return brokerMutex.runExclusive(async () =>
     chainflip.tx.bitcoinIngressEgress
-      .markTransactionAsTainted(txId)
+      .markTransactionAsTainted(memoryRepresentationTxId)
       .signAndSend(broker, { nonce: -1 }, handleSubstrateError(chainflip)),
   );
 }
 
 /**
- * Pauses or resumes the bitcoin block production. We send a command to the docker container to start or stop mining blocks.
+ * Submit a post request to the deposit-monitor, with error handling.
+ * @param portAndRoute Where we want to submit the request to.
+ * @param body The request body, is serialized as JSON.
+ */
+async function postToDepositMonitor(portAndRoute: string, body: string | object) {
+  return axios
+    .post('http://127.0.0.1' + portAndRoute, JSON.stringify(body), {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      timeout: 5000,
+    })
+    .then((res) => res.data)
+    .catch((error) => {
+      let message;
+      if (error.response) {
+        message = error.response.data + ' (' + error.response.status + ')';
+      } else {
+        message = error;
+      }
+      throw new Error('Request to deposit monitor (' + portAndRoute + ') failed: ' + message);
+    });
+}
+
+/**
+ * Set the mockmode of the deposit monitor, controlling how it analyses incoming transactions.
  *
- * @param pause - Whether to pause or resume the block production.
- * @returns - Whether the command was successful.
+ * @param mode Object describing the mockmode we want to set the deposit-monitor to,
+ * Possible example values are: `"Manual"`, `{Deterministic: {score: 1.0, incomplete_probability: 0.5}}`,
+ * `{Random: { min_score: 0.0, max_score: 10.0 , incomplete_probability: 0.5 }}`.
  */
-function pauseBtcBlockProduction(pause: boolean): boolean {
-  try {
-    execSync(
-      pause
-        ? 'docker exec bitcoin rm /root/mine_blocks'
-        : 'docker exec bitcoin touch /root/mine_blocks',
+async function setMockmode(mode: string | object) {
```

Comment:
We could have a type for this that's more specific than "object", also allows typescript to help us a bit more ith type checking

Reply:
True, I can create a type for this.